### ISSUE TITLE
Weird error: virtualenv not found. Succeeds on 2nd try.

### ISSUE THREAD

**Issue by mjmare** [Fizzadar/pyinfra#567]
## Describe the bug
I have a deploy that creates a virtual env and then proceed to install packages in to it. 
The weird thing is that when I run the deploy for the first time it fails. When I rerun the deploy (without modifications!) it succeeds. 
Error:
```
--> Starting operation: Pip/Packages (['setuptools', 'RPi.GPIO', 'adafruit-blinka'], latest=True, requirements=/home/pi/card_reader/requirements.txt, virtualenv=/home/pi/venv)
    [reader01.local] sh: 1: virtualenv: not found
    [reader01.local] Error

```

I have had this on all 4 systems (running RaspberryPi OS lite) that I have configured 

## To Reproduce
The relevant part of the deploy:
```
pip.virtualenv(
    name='Create a virtualenv',
    path='/home/pi/venv',
    python='/usr/bin/python3',
    venv=True,
    sudo=True,
)

# Python packages
pip.packages(name='Install python packages',
             packages=host.data.PYTHON_PACKAGES,
             latest=True,
             requirements=f'{host.data.DEPLOY_DIR}/requirements.txt',
             pip='pip',
             virtualenv='/home/pi/venv',
             sudo=True)
```
## Expected behavior
To work the first time ;-)

## Meta
--> Support information:
    System: Darwin
      Platform: macOS-11.2.3-x86_64-i386-64bit
      Release: 20.3.0
      Machine: x86_64
    pyinfra: v1.3.10
    Executable: /Users/mjm/Dropbox/Devel/ACHN/inuit_cardreader/venv/bin/pyinfra
    Python: 3.9.2 (CPython, Clang 12.0.0 (clang-1200.0.32.29))

**Comment by Fizzadar** [comment]
I've figured this out! This is happening because the `pip.packages` operation is also trying to create the virtualenv, but using `virtualenv` instead of `python -m venv`. This happens because when preparing the operations the virtualenv doesn't exist, so both operations will try to create it.

This is a [limitation of the two phase deploy](https://docs.pyinfra.com/en/1.x/deploy_process.html#limitations), and there's ongoing work to [fix this by making operations properly idempotent](https://github.com/Fizzadar/pyinfra/pull/525).

For an immediate fix you could combine the operations like so:

```py
pip.packages(name='Install python packages',
             packages=host.data.PYTHON_PACKAGES,
             latest=True,
             requirements=f'{host.data.DEPLOY_DIR}/requirements.txt',
             pip='pip',
             virtualenv='/home/pi/venv',
             virtualenv_kwargs={'venv': True},  # this is passed into the `pip.virtualenv` operation
             sudo=True)
```

**Comment by mjmare** [comment]
Thanks for the workaround. Looking forward to real idempotency.

**Comment by Fizzadar** [comment]
This has now been fixed & released in `v1.4`!
